### DISCUSSION TITLE
[@types/react-dom] no longer supports ReactDOM.createRoot()

### DISCUSSION THREAD

**Discussion by shallow-alchemy** [DefinitelyTyped/DefinitelyTyped#59669] [Issues with a @types package] [answered]
@types/react-dom (as of at least v. 17.0.14) no longer supports `ReactDOM.createRoot`. v. 17.0.11 did. The following error is thrown:

```
TS2339: Property 'createRoot' does not exist on type 'typeof import("/myapp/node_modules/@types/react-dom/index")'
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
`createRoot` is no longer part of `react-dom` but `react-dom/client`.

**Reply by shallow-alchemy** [reply]
Thank you. That should not have been a patch version change then but I can confirm it works

**Reply by eps1lon** [reply]
The contract of the `/next` types was that they always support the latest `@next` release tag of React. That contract was not broken.
